locked-asset/simple-lock/src/proxy_farm.rs

```diff
@@ -177,6 +181,11 @@ pub trait ProxyFarmModule:
             &proxy_farm_token_attributes,
         );
 
+        self.send()
+            .direct_non_zero_esdt_payment(&caller, &enter_farm_result.reward_tokens);
```

Comment:
One multiTransfer call would be preferable.

Reply:
Most of the time, this is going to be a single payment, as rewards are more often than not 0 on enter. I'd rather keep it as mostly single payment and sometimes double payment, rather than add the overhead of multi-transfer for 1 payment, but it's your call.